Code/GraphMol/MolDraw2D/test1.cpp

```diff
@@ -4533,6 +4533,12 @@ void test23JSONAtomColourPalette() {
     TEST_ASSERT(text.find("fill:#3366E5") != std::string::npos);
     TEST_ASSERT(text.find("fill:#E59900") != std::string::npos);
 #endif
+    std::regex regex(
+        R"regex(path d='M \d+.\d+,\d+.\d+ L \d+.\d+,\d+.\d+ L \d+.\d+,\d+.\d+.*)regex");
+    std::ptrdiff_t const match_count(
+        std::distance(std::sregex_iterator(text.begin(), text.end(), regex),
+                      std::sregex_iterator()));
```

Comment:
might as well hide a little bit of the horribleness:
```suggestion
    auto match_count = 
        std::distance(std::sregex_iterator(text.begin(), text.end(), regex),
                      std::sregex_iterator());
```

Reply:
I have mixed feelings about auto - it's not always clear what exactly you're getting and how to use it subsequently,  particularly in the case of pointers and references where the downstream syntax is different.  However, as you allude to in this case, std:::ptrdiff_t is hardly revealing, either.  CLion translates it as 'long'.

Reply:
Agreed about auto sometimes making things more difficult. I definitely would be a lot less enthusiastic about it if I didn't use an IDE